src/Razor/src/Microsoft.AspNetCore.Razor.ExternalAccess.OmniSharp/Project/AbstractOmniSharpProjectSnapshotManagerChangeTrigger.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+#nullable disable
+
+using Microsoft;
+
+namespace Microsoft.AspNetCore.Razor.ExternalAccess.OmniSharp.Project;
+
+internal abstract class AbstractOmniSharpProjectSnapshotManagerChangeTrigger
```

Comment:
This used to be `IOmniSharpProjectSnapshotManagerChangeTrigger`. I turned it into an abstract class because I wanted to keep `OmniSharpProjectSnapshotManagerBase` internal but when you put an internal member on an interface you can't implement it because the method has to be public. I don't particularly like my solution, so I'm hoping someone has a better idea.

Reply:
Without knowing why some things are internal, and some public, I'm not sure I can offer any ideas